Sources/TuistCache/Cache/URL+Cache.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+import TuistCore
+
+enum CacheURLError: LocalizedError {
+    case missingCloudConfig
+    case incorrectCloudConfig
+}
+
+extension URL {
+    static func apiCacheURL(hash: String, config: Config, contentMD5: String? = nil) throws -> URL {
+        guard let cloudConfig = config.cloud else { throw CacheURLError.missingCloudConfig }
+        guard var urlComponents = URLComponents(url: cloudConfig.url, resolvingAgainstBaseURL: false) else {
```

Comment:
At this point, we could assume the URL has already been validated so there's no need to introduce this check here. It should be safe to force-unwrap the value.